### PR TITLE
Edge Agent: Cleanup unused docker components

### PR DESCRIPTION
**Pull request by and-rewsmith**
TLDR: CI tests were failing but we can just remove the tests as they aren't needed. And we can also remove other related components in EdgeAgent that are unused as due diligence cleanup.

CI tests were failing due to `Docker.Dotnet` dependency not being compatible with new docker version. We were previously using package:
```
<PackageReference Include="Docker.DotNet" Version="3.125.2" />
```

This package stops working with docker version:
```
20.10.18+azure-2
```

It takes an updated `Docker.Dotnet` version to fix the issue. However the updated `Docker.Dotnet` version changes some key apis we depend on. There are multiple differences, but most problematic one is that we now can no longer return a `Stream` of logs but a `Docker.Dotnet.MultiplexedStream`. This difference introduces a problem because `edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/RuntimeInfoProvider.cs` is implementing an interface that returns a `Stream`.  As such, our options are to either change the interface or convert the `MultiplexedStream` to a `Stream`.

Taking a step back though, the affected CI tests and this whole package upgrade problem occur in code that is not being used in production. It is a leftover from a time when EdgeAgent supported running in docker mode without EdgeAgent. For my prior work on EdgeAgent's ModuleUpdateMode feature, I confirmed that the devtools do not make use of this docker mode so some  key pieces of docker mode are already deleted from EdgeAgent src folder. 

The affected CI tests (`edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Docker.Test/RuntimeInfoProviderTest.cs`) can be safely removed as this is testing the docker `RuntimeInfoProvider` which itself is not used and can be deleted So to summarize the main conceptual cleanup deletions:
1. Remove `RuntimeInfoProviderTest.cs`
2. Remove docker `RuntimeInfoProvider.cs` (keeping the one for edgelet)
3. Remove all other files that are now unused due to the above deletions.
4. Also refreshed local CI variables to make it run locally on dev boxes 

Tested in CI run:
https://msazure.visualstudio.com/One/_build/results?buildId=64122182&view=results

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [x] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/Microsoft.Azure.Devices.Edge.Agent.Docker.csproj [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="5.0.0" />
-    <PackageReference Include="Docker.DotNet" Version="3.125.2" />
+    <PackageReference Include="Docker.DotNet" Version="3.125.12" />
```
Comment (damonbarry) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/Microsoft.Azure.Devices.Edge.Agent.Docker.csproj:13]: Do we need to update Docker.DotNet here? In fact, so we still need Microsoft.Azure.Devices.Edge.Agent.Docker?

Comment (and-rewsmith) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/Microsoft.Azure.Devices.Edge.Agent.Docker.csproj:13]: Earlier I tried removing the dependency altogether. But there are still things that use it in Edge.Agent.Docker (which is still needed because components are depended on from things in AgentModule / EdgeletModule / etc.). I made a pass through Edge.Agent.Docker though to get rid of all not-needed components.

For the dependency version, I went back and forth about it. On one hand leaving it to what it was will minimize changes and tests still pass. But on the other hand we know it isn't compat with new docker versions which makes me suspect. My intuition is that bumping it is the safe bet since tests still pass.

File: edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test.Common/settings/base.json [hunk 2]
```diff
@@ -13,11 +13,11 @@
...
-  "dockerHostUrl": "http://localhost:2375",
+  "dockerHostUrl": "unix://var/run/docker.sock",
   "GatewayHostname": "127.0.0.1",
   "GatewayDeviceId": "device1",
   "StressTest_MessagesCount_SingleSender": "10000",
   "StressTest_MessagesCount_MultipleSenders": "5000",
   "StressTest_MessagesCount_MultipleSendersMultipleReceivers": "5000",
   "Tests_StartEdgeHubService": true
-}
+}
```
Comment (and-rewsmith) [inline edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test.Common/settings/base.json:23]: @damonbarry FYA I updated the client id and thumbprint to make it so that we can run CI tests locally without interacting with these files. IIUC windows doesn't even work so I can go ahead and remove the settings/windows.json.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/Microsoft.Azure.Devices.Edge.Agent.Docker.csproj (hunk 1 above)
Comment (damonbarry) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/Microsoft.Azure.Devices.Edge.Agent.Docker.csproj:13]: Makes sense.

**Review by damonbarry — APPROVED** [review]

**Review by gauravIoTEdge — APPROVED** [review]
